### ISSUE TITLE
fix: Nuxt DevTools client injection failed on Chrome Beta

### ISSUE THREAD

**Issue by martinszeltins** [nuxt/devtools#723]
### 🐛 The bug

The Nuxt Devtools are not working on the latest Chrome Beta version. It works in all other browsers including Chrome stable but not in Chrome Beta (`Version 129.0.6668.29 (Official Build) beta (64-bit)`)

This is with a completely new / fresh installation of Nuxt.

````
client.mjs:30 Nuxt DevTools client injection failed
iframe.onload @ client.js?v=3232c809:111
20:36:13.672 client.mjs:30 undefined
````

Related: https://github.com/nuxt/devtools/issues/488#issuecomment-2340429751

Screenshot:
![devtools](https://github.com/user-attachments/assets/c427d870-3dfe-410f-ba67-fd552f7f0fd0)

### 🛠️ To reproduce

https://stackblitz.com/edit/nuxt-starter-vguwcu?file=nuxt.config.ts

### 🌈 Expected behavior

It should work.

**Comment by antfu** [comment]
Shouldn't that be considered as a Chrome Beta bug? It's working on Chrome Stable. I am not sure what we can do here

**Comment by martinszeltins** [comment]
> Shouldn't that be considered as a Chrome Beta bug? It's working on Chrome Stable. I am not sure what we can do here

Can we investigate why the Nuxt DevTools are not opening in Chrome Beta? Maybe the devtools can be patched to also work in Chrome Beta?

**Comment by antfu** [comment]
I don't think we will have the bandwidth to test every Beta version. In this case it looks more like a Chrome bug that they should fix instead. Or we would count on the community to help fix if it's really a Nuxt DevTools issue (PR welcome!).

Marking this as closed for now. Thanks.

**Comment by jon-lbm** [comment]
> I don't think we will have the bandwidth to test every Beta version. In this case it looks more like a Chrome bug that they should fix instead. Or we would count on the community to help fix if it's really a Nuxt DevTools issue (PR welcome!).
> 
> Marking this as closed for now. Thanks.

I'm actually seeing the same issue on my end and I'm using Version 129.0.6668.42 (Official Build) (arm64)

**Comment by joaolisboa** [comment]
Same issue on my end. On Chromium Engine Version 129.0.6668.42

**Comment by robert-hoffmann** [comment]
Except this is not a beta anymore ..remember chrome moves fast

My dev tools are empty now with chrome 129.0.6668.59 (official)

**Comment by robert-hoffmann** [comment]
> I don't think we will have the bandwidth to test every Beta version. In this case it looks more like a Chrome bug that they should fix instead. Or we would count on the community to help fix if it's really a Nuxt DevTools issue (PR welcome!).
> 
> Marking this as closed for now. Thanks.

Chrome beta moves pretty fast, canary i wouldn't worry about, but something that is broken in beta, is probably gonna hit your user base pretty fast in the official version

I'd include chrome beta as an official testbed for all things nuxt
(food for thought)

**Comment by antfu** [comment]
Now, I can reproduce it with v129:

It looks like really a Chrome bug, but I haven't figured out the root cause yet. But it seems like Vue Router gets mismatched in Chrome with the wrong fullPath: 

<img width="1321" alt="Screenshot 2024-09-19 at 12 31 10" src="https://github.com/user-attachments/assets/a5d951a4-2c41-46ca-b5e3-7711c6d2e074">

Will keep investigating

**Comment by robert-hoffmann** [comment]
Maybe something here...

https://chromestatus.com/roadmap

**Comment by antfu** [comment]
When down, it was caused by a circular import in our client bundle. Somehow Chrome v129 failed to resolve and hangs (dynamic `import(. '/foo.js')` never get resolved). I might need to strim the bundle down to a minimal repro so I can report to Chrome

**Comment by antfu** [comment]
Here is the repro: https://github.com/antfu/repro-chrome-v129 - Now I am very confident it's a chrome bug. I reported to Chrome already, let's see how they response, if not I wound try to report to Chromium later.

**Comment by Moonlight63** [comment]
Same bug in MS Edge

**Comment by antfu** [comment]
Update: it should be fixed in the next version https://x.com/_shu/status/1837009344153669759

In the meantime, I will check our bundle to see if we can workaround it before that

**Comment by skrabbe001** [comment]
> In the meantime, I will check our bundle to see if we can workaround it before that

Which thread should we follow to see if a workaround has been implemented? I see you closed this thread.

**Comment by robert-hoffmann** [comment]
> Which thread should we follow to see if a workaround has been implemented? I see you closed this thread.

You'll need to make sure

    "@nuxt/devtools": "^1.5.0"
    
 is in your devDependencies, and run pnpm install (maybe with --force), to get it running

**Comment by skrabbe001** [comment]
Nice. I saw [manniL](https://github.com/manniL) wrote the following on Nuxt discord, which may be relevant for other readers of this thread:

>when you or a dependency uses an await before imports, that's what the Chrome bug

**Comment by MickL** [comment]
Had the same error suddenly. Deleted the bun lockfile, which then installed `1.5.0` and it works again!

**Comment by thiagonunesbatista** [comment]
> > Which thread should we follow to see if a workaround has been implemented? I see you closed this thread.
> 
> You'll need to make sure
> 
> ```
> "@nuxt/devtools": "^1.5.0"
> ```
> 
> is in your devDependencies, and run pnpm install (maybe with --force), to get it running

This solution worked for me too, I am not using the Devtools module anymore. I installed the devtools manually with the npm package

**Comment by MABottiglieri** [comment]
same issue for me with 129.0.6668.90 (official) (a 64 bit)

**Comment by blqke** [comment]
> same issue for me with 129.0.6668.90 (official) (a 64 bit)

@MABottiglieri Even with `@nuxt/devtools` to `1.5.2` ?

**Comment by hacknug** [comment]
> > same issue for me with 129.0.6668.90 (official) (a 64 bit)
> 
> [@MABottiglieri](https://github.com/MABottiglieri) Even with `@nuxt/devtools` to `1.5.2` ?

Yeah. Same thing happening to me with Arc and v1.5.2.

Version numbers for Arc and Chromium are:
> Versión 1.63.0 (54427)-RC
> Chromium Engine Version 129.0.6668.90

**Comment by robert-hoffmann** [comment]
Min works fine here on chrome: 129.0.6668.90, with

devDependencies:
"@nuxt/devtools": "^1.5.2"

Do remember, you need to press: Shift + Alt + D
to open it

**Comment by jamols09** [comment]
Mine does not even appear when trying to run dev mode.
![Image](https://github.com/user-attachments/assets/acf3b6fe-3740-4439-b3c3-0e44ca86928a)
